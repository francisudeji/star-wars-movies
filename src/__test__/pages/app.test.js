import React from 'react'
import { render } from '@testing-library/react'
import { ReactQueryCacheProvider, QueryCache } from 'react-query'
import { Hydrate } from 'react-query/hydration'
import Home from '../../pages/index'

const queryCache = new QueryCache()
queryCache.setQueryData('films', [])

export const props = {
  Component: Home,
  pageProps: {
    dehydratedState: {}
  }
}

function Component() {
  return (
    <div>
      <h1>Star Wars Films</h1>
      <input type="text" role="search" />
    </div>
  )
}

export function App() {
  return (
    <ReactQueryCacheProvider queryCache={queryCache}>
      <Hydrate state={props.dehydratedState}>
        <Component {...props} films={[]} />
      </Hydrate>
    </ReactQueryCacheProvider>
  )
}

test('Home page renders without errors', () => {
  const utils = render(<App {...props} />)

  expect(App).toMatchSnapshot()
  expect(utils.getByText(/star wars films/i)).toBeInTheDocument()
  expect(utils.getByRole(/search/i)).toBeInTheDocument()
})
