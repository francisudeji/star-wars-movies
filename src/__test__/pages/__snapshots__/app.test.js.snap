// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders without errors 1`] = `[Function]`;
